os/mm/umm_heap/umm_zalloc.c

```diff
@@ -158,6 +162,9 @@ static void *heap_zalloc(size_t size, int s, int e, size_t retaddr)
 
 FAR void *zalloc(size_t size)
 {
+	if (size == 0) {
```

Comment:
mm_malloc and mm_realloc have the same conditional inside so that there are two duplicated checking. We need to remove one.

Reply:
Yes, I will change mm_xalloc in other PR not to check the size which is smaller than 1.
Thanks.